src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,33 @@
+import React, { FC, useEffect, useState } from 'react';
+import cn from 'classnames';
+
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: React.Dispatch<React.SetStateAction<Todo | null>>;
+};
+
+export const TodoModal: FC<Props> = ({ selectedTodo, setSelectedTodo }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const { id, title, completed, userId } = selectedTodo;
+
+  useEffect(() => {
+    getUser(userId)
+      .then(setUser)
+      .finally(() => setIsLoading(false));
+  }, [userId]);
```

Comment:
You are directly interacting with the DOM in the useEffect hook. This is not recommended in React as it can lead to bugs and makes the code harder to understand. Instead, you should use React's state and lifecycle methods to handle any side effects. In this case, you should move the fetching of the user data into a separate function and call that function from within the useEffect hook.